test/cross/cross_test.go

```diff
@@ -0,0 +1,201 @@
+	WorkflowKind        = "Workflow"
+	WorkflowRuntimeKind = "WorkflowRuntime"
+)
+
+func GetSampleWorkflowRuntime() *serverlessv1alpha1.WorkflowRuntime {
+	return &serverlessv1alpha1.WorkflowRuntime{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       WorkflowRuntimeKind,
+			APIVersion: APIVersion,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "simple",
+			Namespace: "default",
+		},
+		Spec: serverlessv1alpha1.WorkflowRuntimeSpec{
+			Replicas: 2,
+		},
+		Status: serverlessv1alpha1.WorkflowRuntimeStatus{
+			Instances: serverlessv1alpha1.Instances{
+				"caller": {
+					Status: serverlessv1alpha1.InstanceStatus{
+						HostIP: "127.0.0.1",
+						PodIP:  "127.0.0.1:8080",
+					},
+					ProcessRuntimes: serverlessv1alpha1.ProcessRuntimes{
+						"simple_start": {
+							Number: 1,
+						},
+						"simple_mid": {
+							Number: 1,
+						},
+					},
+				},
+				"callee": {
+					Status: serverlessv1alpha1.InstanceStatus{
+						HostIP: "127.0.0.1",
+						PodIP:  "127.0.0.1:9090",
+					},
+					ProcessRuntimes: serverlessv1alpha1.ProcessRuntimes{
+						"simple_branch_1": {
+							Number: 1,
+						},
+						"simple_branch_2": {
+							Number: 1,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func GetSampleWorkflow() *serverlessv1alpha1.Workflow {
+	return &serverlessv1alpha1.Workflow{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: APIVersion,
+			Kind:       WorkflowKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "simple",
+			Namespace: "default",
+		},
+		Spec: serverlessv1alpha1.WorkflowSpec{
+			Spec: []serverlessv1alpha1.Flow{
+				{
+					Name:       "simple_start",
+					Function:   "simple_start",
+					Outputs:    []string{"simple_mid"},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       serverlessv1alpha1.Start,
+				},
+				{
+					Name:       "simple_mid",
+					Function:   "simple_mid",
+					Outputs:    []string{"simple_branch_1", "simple_branch_2"},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       "",
+				},
+				{
+					Name:       "simple_branch_1",
+					Function:   "simple_branch_1",
+					Outputs:    []string{},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       serverlessv1alpha1.End,
+				},
+				{
+					Name:       "simple_branch_2",
+					Function:   "simple_branch_2",
+					Outputs:    []string{},
+					Conditions: []*serverlessv1alpha1.Condition{},
+					Statement:  serverlessv1alpha1.Direct,
+					Role:       serverlessv1alpha1.End,
+				},
+			},
+		},
+		Status: serverlessv1alpha1.WorkflowStatus{},
+	}
+}
+
+func DumpConfig(object runtime.Object, fileName string) error {
+	scheme := runtime.NewScheme()
+	_ = serverlessv1alpha1.AddToScheme(scheme)
+	serializer := json.NewSerializerWithOptions(yaml.DefaultMetaFactory, scheme, scheme, json.SerializerOptions{true, false, false})
```

Comment:
Add the key name of SerializerOptions

Reply:
I still don't understand the meaning of Typer even though I read the comment ...